Add support for the Avnet MaaXBoard (seL4_tools)

mjenkcapgemini opened the pull request:
This pull request is a call for feedback and comment on an initial implementation for supporting seL4 on the Avnet MaaXBoard, with the intention of the changes below being merged back to the main seL4_tools repository. This pull request is dependent on another pull request which provides the core support for the MaaXBoard in the seL4 repository; this can be found [here](https://github.com/seL4/seL4/pull/829).

This pull request contains the following changes:
* The creation of a platform_init file for the MaaXBoard, performing the necessary resets for initialising the board.
* Renaming of the specific `KernelPlatformImx8mq-evk` platform from `cmake-tool/helpers/application_settings.cmake`, creating a common `KernelPlatformImx8mq` platform, which can be used by both the imx8mq-evk evaluation board and the Avnet MaaXBoard.

seL4test has been tested with these changes only on the MaaXBoard itself, but all tests run as expected resulting in a `All is well in the universe` message. No other platforms have been tested.

Test with: seL4/seL4#829, seL4/util_libs#129

axel-h commented:
Thar also looks ok, please squash the commits.

axel-h commented:
Thar also looks ok, please squash the commits.

axel-h commented on cmake-tool/helpers/application_settings.cmake:
```diff
@@ -35,9 +36,9 @@ function(ApplyData61ElfLoaderSettings kernel_platform kernel_sel4_arch)
+    if((KernelPlatformImx8mm-evk OR KernelPlatImx8mq) AND KernelSel4ArchAarch32)
         set(ElfloaderArmV8LeaveAarch64 ON CACHE BOOL "" FORCE)
-        # This applies to imx8mm and imx8mq when in aarch32 configuration
+        # This applies to imx8m, imx8mq and maaxboard when in aarch32 configuration
```
```suggestion
        # This applies to imx8m, imx8mq (EVK and MaaxBoard) when in aarch32 configuration
```

I think we can leave `KernelPlatImx8mq` here, as this seems a general i.MX8MQ thing in the end. Just in the comment it should be a bit more clear, where this MaaxBoard belongs to. In the context of this PR it's obvious, but if somebody with a different context reads this comment one day, it might look odd.

mjenkcapgemini replied:
Thanks, fixed.

axel-h commented on cmake-tool/helpers/application_settings.cmake:
```diff
@@ -35,9 +36,9 @@ function(ApplyData61ElfLoaderSettings kernel_platform kernel_sel4_arch)
+    if((KernelPlatformImx8mm-evk OR KernelPlatImx8mq) AND KernelSel4ArchAarch32)
         set(ElfloaderArmV8LeaveAarch64 ON CACHE BOOL "" FORCE)
-        # This applies to imx8mm and imx8mq when in aarch32 configuration
+        # This applies to imx8m, imx8mq and maaxboard when in aarch32 configuration
```
```suggestion
        # This applies to imx8m, imx8mq (EVK and MaaxBoard) when in aarch32 configuration
```

I think we can leave `KernelPlatImx8mq` here, as this seems a general i.MX8MQ thing in the end. Just in the comment it should be a bit more clear, where this MaaxBoard belongs to. In the context of this PR it's obvious, but if somebody with a different context reads this comment one day, it might look odd.

Ivan-Velickovic replied:
This should also say "imx8mm" instead of "imx8m"

mjenkcapgemini commented:
> That also looks ok, please squash the commits.

Addressed review comments concerning the `cmake-tool/helpers/application_settings.cmake` comment, then squashed the commits.
